### ISSUE TITLE
Application crashing in production in particular devices

### ISSUE THREAD

**Issue by surajQuitsure** [gbumps/react-native-screenguard#24]
As per the google crashlytics, while registering the ScreenGuard module application is crashing in some devices.

Devices: Motorola edge 40 - Android 13, Nothing Phone(1) - Android 13, etc.

My implementation of ScreenGuard in react native: 
```
  import { useColorScheme } from 'react-native';

  const isDarkMode = useColorScheme();

  useEffect(() => {
    const screenCaptureColor = isDarkMode === "dark" ? Colors.darker : Colors.lighter
    ScreenGuardModule.register(screenCaptureColor, (_) => {
      console.log("ScreenGuradModule registered")
    });

    //handling android backpress
    const onbackPress = async () => {
      navigation.pop();
    };

    const backHandler = BackHandler.addEventListener(
      'hardwareBackPress',
      onbackPress,
    );

    return async () => {
      backHandler.remove();
      ScreenGuardModule.unregister()
    };
  }, []);
```
  
  As per the crashlytics reports are:

```
  ScreenGuardModule.lambda$deactivateShield$2
  java.lang.NullPointerException
  java.util.Objects.requireNonNull (Objects.java:220)
 com.screenguard.ScreenGuardModule.lambda$deactivateShield$2 (ScreenGuardModule.java:6)
```

**Comment by gbumps** [comment]
Can you give me more infos on react native version you are using and version dependency of `react-native-screenguard`

**Comment by surajQuitsure** [comment]
Yes, here is more information:

```
"react-native": "0.70.6"
"react-native-screenguard": "^0.2.4"
```

**Comment by gbumps** [comment]
Hmm, I understand now. Will have a fix later on.

**Comment by surajQuitsure** [comment]
Can you give me the quick fix so that I can apply that code in node modules, because the application I'm working on has large userbase, if the crashes persists then this application will have major hit on the ranking in playstore

**Comment by gbumps** [comment]
ok, here's work around on the native Android side: 

Check  `ScreenGuardModule.java` and fix the method `deactivateShield` like this:

`@ReactMethod
    public void deactivateShield() {
        try {
            mHandlerBlockScreenShot.post(() -> Objects.requireNonNull(
                    getReactApplicationContext().getCurrentActivity()
            ).getWindow().clearFlags(WindowManager.LayoutParams.FLAG_SECURE));
            mHandlerBlockScreenShot = null;
        } catch (Exception e) {
            e.printStackTrace();
        }
    }`

**Comment by surajQuitsure** [comment]
I'll add this solution and watch the crashlytics for a while and then will let you know

**Comment by gbumps** [comment]
Mark as done, closing. Feel free to reopen if issue occurred again.

**Comment by gbumps** [comment]
This issue has been resolved at ver `0.3.2`, please install and rebuild,  thanks!

**Comment by surajQuitsure** [comment]
Thanks!
